gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala

```diff
@@ -224,28 +224,15 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
         }
         // according to the substrait plan order
         plan match {
-          case shj: HashJoinLikeExecTransformer =>
```

Comment:
The matching and handling for `HashJoinLikeExecTransformer` should be kept. Since it may have no `children` for using your unified statement: `t.children.foreach()`.
In last comment, I meant we can put `match/case` block in line#220 and line226 into single one `match/case` block.
After second thought, I think we can just make the changes for below part in this PR. Please ignore unifying the catch block I said. Thanks!

```
        // according to the substrait plan order
        plan match {
          case shj: HashJoinLikeExecTransformer =>
            transformChildren(shj.streamedPlan, basicScanExecTransformers)
            transformChildren(shj.buildPlan, basicScanExecTransformers)
          case t: TransformSupport =>
            t.children
              .foreach(transformChildren(_, basicScanExecTransformers))
        }
```

Reply:
Sorry to bother you again. I do not understand why the `HashJoinLikeExecTransformer` can not be unified by `t.children.foreach()`. Since the `HashJoinLikeExecTransformer#children` will return the left and right. I think it's equivalent to your block.

Reply:
Correct. Thanks for your reminding! Please rebase your code.

Reply:
Rebased, thanks for your review!